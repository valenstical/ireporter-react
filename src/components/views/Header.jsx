import React from 'react';
import { Link } from 'react-router-dom';

/**
 * The header component
 * @returns {object} The generated JSX object
 */
export default function Header() {
  return (
    <nav className="navigation">
      <div className="wrapper">
        <div className="clear-float">
          <Link to="/" className="nav-brand text-brand float-left">iReporter</Link>
          <ul className="nav-section float-right nav-shown nav-hidden transition" id="nav-section">
            <li data-pg-name="Nav Item">
              <Link to="/">Home</Link>
            </li>
            <li data-pg-name="Nav Item">
              <Link to="/login">Sign in</Link>
            </li>
            <li data-pg-name="Nav Item" className="btn-cta">
              <Link to="/sign-up" className="btn-brand transition">Report a case</Link>
            </li>
          </ul>
          <button className="menu-bar" type="button">
            <i className="fa fa-align-justify" />
          </button>
        </div>
      </div>
    </nav>
  );
}
