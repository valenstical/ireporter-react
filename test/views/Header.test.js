import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../src/components/views/Header';

describe('<Header>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Header user={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
