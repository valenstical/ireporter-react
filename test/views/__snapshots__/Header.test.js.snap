// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render without crashing 1`] = `
<nav
  className="navigation"
>
  <div
    className="wrapper"
  >
    <div
      className="clear-float"
    >
      <Link
        className="nav-brand text-brand float-left"
        replace={false}
        to="/"
      >
        iReporter
      </Link>
      <ul
        className="nav-section float-right nav-shown nav-hidden transition"
        id="nav-section"
      >
        <li
          data-pg-name="Nav Item"
        >
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>
        </li>
        <li
          data-pg-name="Nav Item"
        >
          <Link
            replace={false}
            to="/login"
          >
            Sign in
          </Link>
        </li>
        <li
          className="btn-cta"
          data-pg-name="Nav Item"
        >
          <Link
            className="btn-brand transition"
            replace={false}
            to="/sign-up"
          >
            Report a case
          </Link>
        </li>
      </ul>
      <button
        className="menu-bar"
        type="button"
      >
        <i
          className="fa fa-align-justify"
        />
      </button>
    </div>
  </div>
</nav>
`;
